pkg/skaffold/jib/jib.go

```diff
@@ -17,64 +17,173 @@ limitations under the License.
 package jib
 
 import (
+	"encoding/json"
 	"os"
 	"os/exec"
+	"path/filepath"
 	"sort"
+	"strings"
+	"sync"
+	"time"
 
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 	"github.com/karrick/godirwalk"
 	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
-func getDependencies(cmd *exec.Cmd) ([]string, error) {
-	stdout, err := util.RunCmdOut(cmd)
-	if err != nil {
-		return nil, err
+// pathMap contains fields parsed from Jib's JSON output
+type pathMap struct {
+	// Build lists paths to build definitions that trigger a call out to Jib to refresh the pathMap, as well as a rebuild, upon changing
+	Build []string
+
+	// Inputs lists paths to build dependencies that trigger a rebuild upon changing
+	Inputs []string
+
+	// Ignore lists paths to files that should be ignored when checking for changes to rebuild
+	Ignore []string
```

Comment:
We should rename this to be _results_ or something that indicates that they are the result of transformation.  Whether they are ignored depends on the consumer — the image-rebuilder should ignore them, but the file-sync should propagate them.

Reply:
Could you explain this a bit more? Not sure I understand, I might be missing some context.

Reply:
_Ignore_ describes the handling of the files, whereas _Inputs_ and _Build_ are categories.  Now that I have jib-sync working, I realize that this _Ignore_ category should be two categories: _Build Results_ (or _Outputs_?) and _Unknown_.

The Skaffold file-watcher looks at the files changed in a workspace/context and determines whether the file changes
  - can be sync'd into a running container (currently only considers the _Inputs_, but should be able to also include _Build Results_),
  - requires the container to be rebuilt and redeployed,
  - or should just be ignored.

Reply:
Ah I see. In that case, I suppose we'll need to make Jib-side changes to support this?

Reply:
> can be sync'd into a running container (currently only considers the Inputs, but should be able to also include _Build Results_),

Just asking for my own understanding: even with _Build Results_, isn't it that we need further categorization because some build results (e.g., optimized .js processed from source, resources generated during build) should be synced whilst some build results (e.g., `.class`) shouldn't?

Reply:
`.class` files do make sense to be sync'd since things like Spring Boot DevTools can trigger a reload.  I'm mixed on trying to further categorize files — although it's possible, I'm not sure how we could leverage it in Skaffold.  I think we would get greater benefit from providing _smart sync rules_ as per #1844.

I think we can rename this field while leaving the Jib-side field unchanged.  We'll reconcile the two when we start on #1844.